src/snowflake/snowpark/__init__.py

```diff
@@ -67,3 +67,35 @@
     WhenNotMatchedClause,
 )
 from snowflake.snowpark.window import Window, WindowSpec
+
+
+def register_sql_magic():
+    try:
+        import IPython
+        from IPython.core.magic import register_cell_magic
+
+        def sql(line, cell):
+            user_ns = IPython.get_ipython().user_ns
+            if "session" in user_ns:
+                session = user_ns["session"]
```

Comment:
how does user_ns get populated with objects? how do we know that `snowflake.snowpark.Session` will always be mapped to `"session"`

Reply:
We don't know before hand, that why we check and if it is not there we notify the user. We can iterate thru all the objects until we find a session object as well.

Reply:
we would either need to document that session object must be named `session` and no other object must be named session, else these changes won't work